templates/cluster-template-eks-ipv6.yaml

```diff
@@ -0,0 +1,87 @@
+---
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: Cluster
+metadata:
+  name: "${CLUSTER_NAME}"
+spec:
+  clusterNetwork:
+    pods:
+      cidrBlocks: ["192.168.0.0/16"]
```

Comment:
Doesn't this need to be an IPv6 CIDR?

Reply:
Good question!! That's actually being ignored. :D But good catch, I will document this behaviour. In case of ipv6 you aren't allowed to set pod cidrs. AWS doesn't allow it.

Reply:
You mean EKS?

Reply:
In case of ipv6 we have to override the network config and pass in ipv6 family. This is done here:

```go
	var netConfig *eks.KubernetesNetworkConfigRequest
	if s.scope.VPC().IsIPv6Enabled() {
		netConfig = &eks.KubernetesNetworkConfigRequest{
			IpFamily: aws.String(eks.IpFamilyIpv6),
		}
	} else {
		netConfig, err = makeKubernetesNetworkConfig(s.scope.ServiceCidrs())
		if err != nil {
			return nil, errors.Wrap(err, "couldn't create Kubernetes network config for cluster")
		}
	}

```

During createCluster. This could be done a bit more friendly, like only removing any pod cidrs but leaving anything else that someone might have set up.

Reply:
Found it!

```
If you specify ipv6, Kubernetes assigns Service
	// and Pod addresses from the unique local address range (fc00::/7). You can't
	// specify a custom IPv6 CIDR block.
```

I will add a validation where I'll disallow setting this field if ipv6 is enabled.

Thanks for reminding me about this @sedefsavas  :)) 👍

Reply:
ACK, thanks for the explanation.

We will probably need a little more complicated template for the unmanaged side, need some kubeadm settings for it as well.
Like CAPZ has here:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/2f36cf2ecbcb6bad3827bc5b9f8421a86f7efa21/templates/cluster-template-ipv6.yaml#L96

Reply:
Ah, interesting. Thanks for pointing me to that. 👍

Reply:
I think we discussed with @richardcase to support EKS clusters first, then unmanaged I think? But then some work went into doing bootstrapping and node handling that technically made it work for unmanaged as well with the cidr splitting. Why am I saying this? Because I think there might be other landmindes maybe for unmanaged clusters that I haven't even started looking at?